dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/db/DbType.kt

```diff
@@ -96,4 +194,213 @@ public abstract class DbType(public val dbTypeInJdbcUrl: String) {
         }
         return connection
     }
+
+    /**
+     * Extracts a value from the ResultSet for the given column.
+     * This method can be overridden by custom database types to provide specialized parsing logic.
+     *
+     * @param [rs] the ResultSet to read from
+     * @param [columnIndex] zero-based column index
+     * @param [columnMetadata] metadata for the column
+     * @param [kType] the Kotlin type for this column
+     * @return the extracted value, or null
+     */
+    public open fun extractValueFromResultSet(
+        rs: ResultSet,
+        columnIndex: Int,
+        columnMetadata: TableColumnMetadata,
+        kType: KType,
+    ): Any? =
+        try {
+            rs.getObject(columnIndex + 1)
+            // TODO: add a special handler for Blob via Streams
+        } catch (_: Throwable) {
+            // TODO: expand for all the types like in generateKType function
+            if (kType.isSupertypeOf(String::class.starProjectedType)) {
+                rs.getString(columnIndex + 1)
+            } else {
+                rs.getString(columnIndex + 1)
+            }
+        }
+
+    /**
+     * Builds a single DataColumn with proper type handling.
+     * Accepts a mutable list to allow efficient post-processing.
+     */
+    public open fun buildDataColumn(
+        name: String,
+        values: MutableList<Any?>,
+        kType: KType,
+        columnMetadata: TableColumnMetadata,
+        inferNullability: Boolean,
+    ): DataColumn<*> {
+        val correctedValues = postProcessColumnValues(values, kType, columnMetadata)
+
+        return DataColumn.createValueColumn(
+            name = name,
+            values = correctedValues,
+            infer = convertNullabilityInference(inferNullability),
+            type = kType,
+        )
+    }
+
+    private fun convertNullabilityInference(inferNullability: Boolean) =
```

Comment:
seems a bit too specific to be a function, don't you think? It's also just used but once